cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/pipeline/AppDeploymentInfo.java

```diff
@@ -48,43 +52,76 @@ public class AppDeploymentInfo {
 
   public AppDeploymentInfo(AppDeploymentInfo info, Location artifactLocation) {
     this(info.artifactId, artifactLocation, info.namespaceId, info.applicationClass, info.appName, info.appVersion,
-         info.configString, info.ownerPrincipal, info.updateSchedules, info.runtimeInfo);
+         info.configString, info.changeSummary, info.owner, info.ownerPrincipal,
+         info.updateSchedules, info.runtimeInfo);
   }
 
   public AppDeploymentInfo(ArtifactId artifactId, Location artifactLocation, NamespaceId namespaceId,
                            ApplicationClass applicationClass, @Nullable String appName, @Nullable String appVersion,
                            @Nullable String configString) {
     this(artifactId, artifactLocation, namespaceId, applicationClass, appName, appVersion, configString, null,
-         true, null);
+         null, null, true, null);
+  }
+
+  public AppDeploymentInfo(ArtifactId artifactId, Location artifactLocation, NamespaceId namespaceId,
```

Comment:
Add a Builder instead of adding all of these constructors. Whenever we want to have a bunch of constructors, it's usually a sign that a Builder would be more appropriate.